sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/EventingTestKit.java

```diff
@@ -0,0 +1,168 @@
+import kalix.javasdk.testkit.impl.EventingTestKitImpl;
+
+import java.time.Duration;
+import java.util.List;
+
+
+public interface EventingTestKit {
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  static EventingTestKit start(ActorSystem system, String host, int port, MessageCodec codec) {
+    return EventingTestKitImpl.start(system, host, port, codec);
+  }
+
+
+  Topic getTopic(String topic);
+
+  /**
+   * Testkit utility to mock broker's topic. Useful when doing integration tests for services that do eventing (in or out) to a broker's topic.
+   */
+  @ApiMayChange
+  interface Topic {
+
+    /**
+     * Waits for predefined amount of time. If a message arrives in the meantime or
+     * has arrived before but was not consumed, the test fails.
+     */
+    void expectNone();
+
+    /**
+     * Waits for given amount of time. If a message arrives in the meantime or
+     * has arrived before but was not consumed, the test fails.
+     *
+     * @param timeout amount of time to wait for a message
+     */
+    void expectNone(Duration timeout);
+
+    /**
+     * Waits and returns the next unread message on this topic. Note the message might have been received before this
+     * method was called. If no message is received, a timeout exception is thrown.
+     *
+     * @return a Message with a ByteString payload
+     */
+    Message<ByteString> expectOneRaw();
+
+    /**
+     * Waits and returns the next unread message on this topic. Note the message might have been received before this
+     * method was called. If no message is received, a timeout exception is thrown.
+     *
+     * @param timeout amount of time to wait for a message
+     * @return a Message with a ByteString payload
+     */
+    Message<ByteString> expectOneRaw(Duration timeout);
+
+    /**
+     * Waits and returns the next unread message on this topic.
+     * Note the message might have been received before this method was called.
+     * If no message is received, a timeout exception is thrown.
+     *
+     * @return message including ByteString payload and metadata
+     */
+    Message<?> expectOne();
+
+    /**
+     * Waits for a specific amount and returns the next unread message on this topic.
+     * Note the message might have been received before this method was called.
+     * If no message is received, a timeout exception is thrown.
+     *
+     * @param timeout amount of time to wait for a message if it was not received already
+     * @return message including ByteString payload and metadata
+     */
+    Message<?> expectOne(Duration timeout);
+
+    /**
+     * Waits and returns the next unread message on this topic and automatically parses
+     * and casts it to the specified given type.
+     *
+     * @param instance class type to cast the received message bytes to
+     * @return a Message of type T
+     * @param <T> a given domain type
+     */
+    <T extends GeneratedMessageV3> Message<T> expectMessageType(Class<T> instance);
```

Comment:
Naming is a bit inconsistent:
```
 - expectOne
 - expectMessageType
 ```
 my proposition:
 ```
  - expectMsg
 - expectMsgType
```
I'm not sure what to do with `expectN` because it's more about the time and not the number of msgs. @octonato suggest sth like `expectWhile` which explains better the intention. Could be renamed to `expectMsgsWhile`

Reply:
In Akka typed testkit we call it `receiveN` for a number and `receive` (because probe already typed) that doesn't really assert any properties of the message.

Reply:
Glad you opened this discussion because I were to open it myself and forgot.

I think another option could be:
```
 - expectOne
 - expectOneTyped
 - expectN
 ```

> I'm not sure what to do with `expectN` because it's more about the time and not the number of msgs.

Not sure I agree here. I think it's as much about the number as about the time, depending on what kind of args you pass in. By default, the rational was to focus on the number of messages and use a big timeout.

Reply:
> I'm not sure what to do with expectN because it's more about the time and not the number of msgs.

My bad, correct, it's about both.

Reply:
I'm fine with any convention as long as it is consistent, my suggestion was based on untyped `akka.testkit.TestProbe`

Reply:
updated now in https://github.com/lightbend/kalix-jvm-sdk/pull/1586/commits/c053feecb06f86a4a0a9699d429da65d9dbb3a69